Initial commit of work

autumnfound opened the pull request:
Not final, but represents ~90% completion state

Signed-off-by: Martin Lowe <martin.lowe@eclipse-foundation.org>

mbarbero commented on pom.xml:
```diff
@@ -0,0 +1,137 @@
+			<groupId>org.gitlab4j</groupId>
+			<artifactId>gitlab4j-api</artifactId>
+			<version>4.14.11</version>
+		</dependency>
```
as far as I can tell, this dependency is not used

mbarbero commented on src/main/java/org/eclipsefoundation/git/eca/helper/CommitHelper.java:
```diff
@@ -0,0 +1,66 @@
+	 * @return the email address in the Signed-off-by footer, or null if none could
+	 *         be found.
+	 */
+	public static String getSignedOffByEmail(Commit c) {
```
Please write unit tests for this method

mbarbero commented on src/main/java/org/eclipsefoundation/git/eca/helper/CommitHelper.java:
```diff
@@ -0,0 +1,66 @@
+	 * @param c commit to validate
+	 * @return true if valid, otherwise false
+	 */
+	public static boolean validateCommit(Commit c) {
```
Please write unit tests for this method

mbarbero commented on src/main/java/org/eclipsefoundation/git/eca/helper/CommitHelper.java:
```diff
@@ -0,0 +1,66 @@
+
+		boolean valid = true;
+		// check current commit data
+		if (c.getHash() == null || c.getParents() == null) {
```
some commits (initals) have no parent

autumnfound commented:
I've added a ton of unit tests, and added secret config fetch code for k8s secret stuff we will need later. Also fixed that I was missing license info in a lot of files. Did a lot of code clean up with some help from Sonar.

mbarbero commented on src/main/java/org/eclipsefoundation/git/eca/config/SecretConfigSource.java:
```diff
@@ -0,0 +1,89 @@
+				secrets.putAll((Map) p);
+
+			} catch (IOException e) {
+				LOGGER.error("Error while reading in secrets configuration file.", e);
```
Is it really a recoverable exception? IMO, you should throw a RuntimeExcepton wrapping `e`after logging

autumnfound replied:
This is a common class across multiple APIs, its a target for extraction to a separate project for consumption as a dependency. WDYT?
